pkg/component/component.go

```diff
@@ -0,0 +1,82 @@
+package component
+
+import (
+	"strings"
+
+	"github.com/improbable-eng/thanos/pkg/store/storepb"
+)
+
+type StoreAPI interface {
```

Comment:
Go wants comments like this for exported functions:
```suggestion
// StoreAPI is ...
type StoreAPI interface {
```

https://blog.golang.org/godoc-documenting-go-code

Reply:
Exaclty this ^